python/tests/expressions/test_expressions.py

```diff
@@ -254,15 +218,15 @@ def test_empty_bind_not_in(table_schema_simple: Schema):
 
 def test_bind_not_in_equal_term(table_schema_simple: Schema):
     bound = BoundNotIn(
-        BoundReference(table_schema_simple.find_field(1), table_schema_simple.accessor_for_field(1)), {literal("hello")}
+        BoundReference[str](table_schema_simple.find_field(1), table_schema_simple.accessor_for_field(1)), {"hello"}
```

Comment:
I don't think that bound references should allow bare values. This creates more changes and doesn't seem necessary.

Reply:
Yes, I think this one was required at some point, but it looks like we're able to remove the type 👍🏻